django/core/management/__init__.py

```diff
@@ -254,14 +254,12 @@ def fetch_command(self, subcommand):
         try:
             app_name = commands[subcommand]
         except KeyError:
-            if os.environ.get("DJANGO_SETTINGS_MODULE"):
-                # If `subcommand` is missing due to misconfigured settings, the
-                # following line will retrigger an ImproperlyConfigured exception
-                # (get_commands() swallows the original one) so the user is
-                # informed about it.
-                settings.INSTALLED_APPS
+            if os.environ.get("DJANGO_SETTINGS_MODULE") and self.settings_exception:
+                sys.stderr.write(self.style.NOTICE(str(self.settings_exception) + "\n"))
+                sys.exit(1)
             elif not settings.configured:
-                sys.stderr.write("No Django settings specified.\n")
+                sys.stderr.write(self.style.NOTICE("No Django settings specified.\n"))
+                sys.exit(1)
```

Comment:
I'm thinking maybe I should remove this `sys.exit(1)` (and the one 3 lines up), and let it check for close matches before exiting. So that we could get this:
```
$ python3 ./manage.py startap
Settings module 'settings' could not be imported.
Unknown command: 'startap'. Did you mean startapp?
Type 'manage.py help' for usage.
```

Reply:
No, the `requires_settings` attribute is only checked below, if the command is found despite there being a settings exception.